src/core/io/NewickConverter.cpp

```diff
@@ -404,32 +512,235 @@ TopologyNode* NewickConverter::createNode(const std::string &n, std::vector<Topo
+        }
+
+        // skip semi-colon
+        if ( char( ss.peek() ) == ';' )
+        {
+            // Avoid infinite loop.
+            throw RbException()<<"Not enough closing parentheses!";
+        }
+    }
+
+    // remove closing parenthesis
+    ss.ignore();
+
+    // read the optional label, checking for EOF = '\377'
+    std::string lbl = "";
+    while ( ss.good() && (c = char( ss.peek() )) != ':' && c != ';' && c != ',' && c != '{' && c != '\377')
+    {
+        lbl += char( ss.get() );
+    }
+    node->setName( lbl );
+
+//    // read the optional branch length
+//    if ( char( ss.peek() ) == ':' )
+//    {
+//        
+//        ss.ignore();
+//            
+//        // the total branch time
+//        std::string time = "";
+//            
+//        // read the node character history
+//        if ( ss.peek() == '{' )
+//        {
+//            std::vector<size_t> states;
+//            std::vector<double> times;
+//            do
+//            {
+//                ss.ignore();
+//                
+//                // read the state
+//                std::string state_str = "";
+//                while ( ss.good() && (c = char( ss.peek() ) ) != ',' && c != '}')
+//                {
+//                    state_str += char( ss.get() );
+//                }
+//                size_t state = StringUtilities::asIntegerNumber( state_str );
+//                states.push_back( state );
 //
-//    // convert to AdmixtureNode*
-//    std::vector<AdmixtureNode*> adm_nodes;
-//    for (size_t i = 0; i < nodes.size(); i++)
-//        adm_nodes.push_back(static_cast<AdmixtureNode*>(nodes[i]));
+//                // ignore the equal sign between parameter name and value
+//                if ( ss.peek() == ',')
+//                {
+//                    ss.ignore();
+//                }
 //
-//    // set up the tree
-//    t->setRoot(adm_nodes[adm_nodes.size()-1]);
+//                // read the parameter name
+//                std::string duration_str = "";
+//                while ( ss.good() && (c = char( ss.peek() ) ) != ';' && c != '}' )
+//                {
+//                    duration_str += char( ss.get() );
+//                }
+//                double duration = atof( duration_str.c_str() );
+//                times.push_back( duration );
+//                time += duration;
 //
-//    // set the branch lengths
-//    //for (size_t i = 0; i < nodes.size(); ++i) {
-//    t->setAgesFromBrlens(brlens);
-//    ;//t->setBranchLength(nodes[i]->getIndex(), brlens[i]);
-//    // }
+//            } while ( (c = char( ss.peek() ) ) == ';' );
 //
-//    // return the tree, the caller is responsible for destruction
-//    return t;
-//}
+//            // ignore the final '}'
+//            if ( (c = char( ss.peek( ) ) ) == '}' )
+//            {
+//                ss.ignore();
+//            }
+//        }
+//    }
+//    else
+//    {
+//        nodes.push_back( node );
+//        brlens.push_back( 0.0 );
+//    }
```

Comment:
Can we just delete this? What is the current treatment of branch lengths in the SIMMAP format? Do we allow them but ignore them?